addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3272,8 +3273,42 @@ export class OdooEditor extends EventTarget {
                 this._applyCommand('indentList', ev.shiftKey ? 'outdent' : 'indent');
             } else if (closestTag === 'TABLE') {
                 this._onTabulationInTable(ev);
-            } else if (!ev.shiftKey) {
+            } else if (ev.shiftKey) {
+                new Set(getTraversedNodes(this.editable)
+                    .map(node => closestBlock(node).firstElementChild)
+                    .filter(node => node && isEditorTab(node) &&
+                        !getAdjacentPreviousSiblings(node, sibling => (
+                            sibling.nodeType !== Node.TEXT_NODE || !/^[\u200B\s]*$/.test(sibling.textContent)
+                        )).length)
+                ).forEach(tab => {
```

Comment:
Ok, that was very heavy but bearable up until `isEditorTab(node)` but then the rest is too much considering all that came before it. I suggest to rewrite this chain for the benefit of common vulgar humans.

While we're at it, at the end we'll have a nice iterable variable and we'll be able to do a `for ... of` on it instead of a `forEach`, as I think a `forEach` of 30 lines is kind of a misuse of the feature IMO... ^^'